ZenPacks/zenoss/LinuxMonitor/transforms.py

```diff
@@ -0,0 +1,127 @@
+            usedPercent = current
+            usedBytes = totalBytes * (current * 0.01)
+            freeBytes = totalBytes - usedBytes
+
+        # Windows (Windows ZenPacks)
+        elif 'FreeMegabytes' in evt.eventKey:
+            freeBytes = int(current) * 1048576
+            usedBytes = totalBytes - freeBytes
+            usedPercent = (usedBytes / float(totalBytes)) * 100.0
+
+        else:
+            return
+
+        # Make a nicer event summary.
+        from Products.ZenUtils.Utils import convToUnits
+        evt.summary = "disk space threshold: %3.1f%% used (%s free)" % (
+            usedPercent, convToUnits(freeBytes))
+        evt.message = evt.summary
+
+transform()
+""".strip()
+
+# MD5 hex digests of /Perf/Filesystem transforms that we'll replace.
+PERF_FILESYSTEM_ORIGINAL_DIGESTS = (
+    "18c01c20b29fef6f0b097a8bea443fa6",  # Zenoss 5.2.4
+    "2b10c421920843e25c0aba6fb09f37d6",  # LinuxMonitor 2.2.3
+)
+
+# Description of transforms we want to install.
+EVENT_CLASS_TRANSFORMS = {
+    "/Perf/Filesystem": {
+        "desired": PERF_FILESYSTEM_TRANSFORM,
+        "original_digests": PERF_FILESYSTEM_ORIGINAL_DIGESTS,
+    },
+}
+
+
+def install_transforms(dmd):
+    """Install all transforms in EVENT_CLASS_TRANSFORMS.
+
+    They will only be installed if the existing transform has not been
+    customized by the user.
+
+    """
+    for eventclass_name, data in EVENT_CLASS_TRANSFORMS.iteritems():
+        try:
+            eventclass = dmd.Events.getOrganizer(eventclass_name)
+        except KeyError:
+            continue
+        else:
+            install_transform(
+                eventclass,
+                data.get("desired", ""),
+                data.get("original_digests", []))
+
+
+def install_transform(eventclass, transform, original_digests):
+    """Install transform for eventclass.
+
+    It will only be installed if the existing transform has not been
+    customized by the user. The original_digests are used to determine if
+    the existing transform has been customized. If the MD5 hex digest of the
+    existing transform matches on of original_digests, the new transform
+    will be installed.
+
+    """
+    existing_transform = eventclass.transform or ""
+
+    # No need to do anything if the transform is already updated.
+    if transform == existing_transform:
+        return
+
+    # Avoid overwriting custom transform, but offer the user enough
+    # information to update it on their own if they so choose.
+    if hashlib.md5(existing_transform).hexdigest() not in original_digests:
+        LOG.warning(
+            "Not updating /Perf/Filesystem transform because it has been customized\n"
+            "\n"
+            "It would have been updated to the following otherwise.\n"
+            "\n"
+            "%s",
+            transform)
+
+        return
+
+    LOG.info("Updating /Perf/Filesystem transform for percentUsed datapoint")
+    eventclass.transform = transform
```

Comment:
I'm showing that **Not updating /Perf/Filesystem transform because it has been customized** message when installing the ZP. I'm not aware of any transform customization on this system.